Multisig/Data/Services/Safe Transaction Service/SignTransactionRequest.swift

```diff
@@ -0,0 +1,80 @@
+//
+//  SignTransactionRequest.swift
+//  Multisig
+//
+//  Created by Andrey Scherbovich on 17.09.20.
+//  Copyright © 2020 Gnosis Ltd. All rights reserved.
+//
+
+import Foundation
+import Web3
+
+struct SignTransactionRequest: JSONRequest {
+    let safe: String
+
+    let to: String?
+    let value: String
+    let data: String?
+    let operation: Int
+    let safeTxGas: String
+    let baseGas: String
+    let gasPrice: String
+    let gasToken: String?
+    let refundReceiver: String
+    let nonce: String
+    let contractTransactionHash: String
+    let sender: String
+    let signature: String?
+
+    enum CodingKeys: String, CodingKey {
+        case sender, to, nonce, safeTxGas, baseGas, gasPrice, gasToken, refundReceiver,
+             contractTransactionHash, data, operation, value, signature
+    }
+
+    enum RequestError: String, Error {
+        case couldNotVerifySafeTxHash
+    }
+
+    init(transaction: Transaction, safeAddress: Address) throws {
+        safe = safeAddress.checksummed
+        to = transaction.to?.description
+        value = transaction.value.description
+        data = transaction.data?.description
+        operation = transaction.operation.rawValue
+        safeTxGas = transaction.safeTxGas.description
+        baseGas = transaction.baseGas.description
+        gasPrice = transaction.gasPrice.description
+        gasToken = transaction.gasToken?.description
+        refundReceiver = transaction.refundReceiver.description
+        nonce = transaction.nonce.description
+        contractTransactionHash = transaction.safeTxHash.description
+
+        let hashToSign = Data(ethHex: contractTransactionHash)
+
+        guard let pkData = try? App.shared.keychainService.data(forKey: KeychainKey.ownerPrivateKey.rawValue),
+              let pk = try? EthereumPrivateKey(pkData.bytes),
+              let sig = try? pk.sign(hash: hashToSign.bytes) else {
+            sender = ""
+            signature = ""
+            return
+        }
+        sender = pk.address.hex(eip55: true)
+
+        let v = String(format:"%02X", sig.v + 27)
```

Comment:
Not sure if understand it from the first reading. Is this a conversion from int to hex? Is there a way to convert the sig obj directly to data and then to hex?

Reply:
ok, I will rewrite it